test/rubygems/test_gem_commands_exec_command.rb

```diff
@@ -216,7 +216,6 @@ def test_gem_with_platform_dependencies
 
   def test_gem_with_platform_and_platform_dependencies
     pend "extensions don't quite work on jruby" if Gem.java_platform?
-    pend "terminates on mswin" if Gem.win_platform? && ruby_repo?
```

Comment:
I'm not sure we can remove this one, since this was a ruby/ruby specific issue that I'm not sure it's related to this? See https://github.com/rubygems/rubygems/pull/6417#discussion_r1119573694 and comments below.

Reply:
It is confusing indeed. Thanks for double checking this!

Reply:
It's running now...

Reply:
I feel like posting a NSFW message, but...

Bunch of failures in ruby/ruby, all because they're being run from the build folder.  JFYI, a long time ago I decided to run all the tests in ruby-loco from the install folder, and I've suggested that ruby/ruby do the same.  Or, there's very little testing that verifies that `make install` makes a usable installation.  And there's plenty of 'hacky' code to make tests run both ways.  Anyway...

Question - many of the `pends` in the existing code used `ENV.key?("GITHUB_ACTIONS")`.  That will skip in any CI, not just ruby/ruby.  The run here verified that the tests work from RubyGems CI.  Also, `ruby_repo?` uses:
```
!ENV["GEM_COMMAND"].nil?
```
Are you confident that the above is never true in the CI here?  I don't know, but I'm can't think of any good way to determine if tests are being run from ruby/ruby.  Ideally, it should work for both local & in GHA.  I'm going to check a few things, do you have any thoughts?

If we want to ignore local testing, `ENV['GITHUB_REPOSITORY']` in GHA here would be 'rubygems/rubygems', in ruby/ruby it would (obviously) be 'ruby/ruby'.  We could use ENV['GITHUB_REPOSITORY'].end_with? "/ruby".  Would that work?  People could set it for local work.  If you're okay with that, should I change `ruby_repo?` or just the ENV test in the pend statements?

Sorry about the mess, but I hate skipping tests when they don't need to be...

Reply:
> Are you confident that the above is never true in the CI here? I don't know, but I'm can't think of any good way to determine if tests are being run from ruby/ruby. Ideally, it should work for both local & in GHA. I'm going to check a few things, do you have any thoughts?

I'm pretty sure we're not setting `ENV["GEM_COMMAND"]` here, yes. So I think that works fine.

I think this PR is fine as is, just remove this particular change so we don't break ruby/ruby CI when this change is backported.

Reply:
Thanks.  Updated everything.  Almost all (or maybe all) the tests fail in ruby/ruby, so rather than the following, which is skipped in **all GHA**:
```ruby
pend if RUBY_PLATFORM.include?("mswin") && ENV.key?("GITHUB_ACTIONS")
```

we're now using:
```ruby
pend "terminates on mswin" if vc_windows? && ruby_repo?
```

Reply:
@deivid-rodriguez

I thought I'd take the updated version from here and apply it to ruby/ruby.  It did not work, so please don't merge.

I may fix tonite, or possibly tomorrow.  I'm -0500.

Reply:
Fixed.  Passsing both here and ruby/ruby